client/state/jetpack-connect/selectors.js

```diff
@@ -21,6 +26,14 @@ const getFlowType = function( state, site ) {
 	return false;
 };
 
+const getJetpackSiteByUrl = ( state, url ) => {
+	const site = getSiteByUrl( state, url );
+	if ( site && ! site.jetpack ) {
+		return false;
```

Comment:
since we are at it, can we make this return "null" instead for false ? so we are consistent with the getSiteByUrl response format